File: sdk-dotnet/src/LittleHorseSDK/BasicExampleConsoleApp/MyWorker.cs

using LittleHorse.Worker.Attributes;
using Microsoft.Extensions.Logging;

namespace BasicExampleConsoleApp
{
    public class MyWorker
    {
        private readonly ILogger? _logger;
        public MyWorker(ILogger? logger = null) {
            _logger = logger;
        }

        [LHTaskMethod("greet")]
        public string Greeting(string name)
        {
            _logger?.LogDebug("Executing task greet");

            return $"Hello, {name}";
        }
    }
}

File: examples/basic/src/main/java/io/littlehorse/examples/MyWorker.java
package io.littlehorse.examples;

import io.littlehorse.sdk.worker.LHTaskMethod;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class MyWorker {

    @LHTaskMethod("charge-credit-card")
    public String greeting(String user, int price) {
        System.out.println("Price: " + price);
        return "hello there, " + user;
    }

}
